app-modules/interaction/src/Filament/Concerns/HasManyMorphedInteractionsTrait.php

```diff
@@ -0,0 +1,123 @@
+use Filament\Infolists\Components\TextEntry;
+
+trait HasManyMorphedInteractionsTrait
+{
+    public function infolist(Infolist $infolist): Infolist
+    {
+        return $infolist
+            ->schema([
+                TextEntry::make('user.name')
+                    ->label('Created By'),
+                Fieldset::make('Details')
+                    ->schema([
+                        TextEntry::make('initiative.name'),
+                        TextEntry::make('driver.name'),
+                        TextEntry::make('division.name'),
+                        TextEntry::make('outcome.name'),
+                        TextEntry::make('relation.name'),
+                        TextEntry::make('status.name'),
+                        TextEntry::make('type.name'),
+                    ]),
+                Fieldset::make('Time')
+                    ->schema([
+                        TextEntry::make('start_datetime')
+                            ->label('Start Time')
+                            ->dateTime(),
+                        TextEntry::make('end_datetime')
+                            ->label('End Time')
+                            ->dateTime(),
+                        TextEntry::make('start_datetime')
+                            ->label('Duration')
+                            ->state(fn ($record) => $record->end_datetime->diffForHumans($record->start_datetime, CarbonInterface::DIFF_ABSOLUTE, true, 6)),
+                    ]),
+                Fieldset::make('Notes')
+                    ->schema([
+                        TextEntry::make('subject'),
+                        TextEntry::make('description'),
+                    ]),
+            ]);
+    }
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->recordTitleAttribute('id')
+            ->columns([
+                IdColumn::make(),
+                TextColumn::make('initiative.name'),
+                TextColumn::make('driver.name'),
+                TextColumn::make('division.name'),
+                TextColumn::make('outcome.name'),
+                TextColumn::make('relation.name'),
+                TextColumn::make('status.name'),
+                TextColumn::make('type.name'),
+                TextColumn::make('start_datetime')
+                    ->label('Start Time')
+                    ->dateTime(),
+                TextColumn::make('end_datetime')
+                    ->label('End Time')
+                    ->dateTime(),
+                TextColumn::make('created_at')
+                    ->state(fn ($record) => $record->end_datetime->diffForHumans($record->start_datetime, CarbonInterface::DIFF_ABSOLUTE, true, 6))
+                    ->label('Duration'),
+                TextColumn::make('subject'),
+                TextColumn::make('description'),
+            ])
+            ->headerActions([
+                CreateAction::make()
+                    ->authorize('create', $this->getOwnerRecord()),
```

Comment:
Does `$this->getOwnerRecord()` reference the correct Model here?

Reply:
Yep ! It's referring correct model here.